Silicon/Bosc/NanHuPkg/Library/PciHostBridgeLib/PciHostBridgeLib.h

```diff
@@ -0,0 +1,33 @@
+/** @file
+  Main Header file for the PciHostBridgeLib
+
+  Copyright (c) 2024, Bosc. All rights reserved.<BR>ved.<BR>
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+#ifndef __PCIHOSTBRIDGELIB_H
+#define __PCIHOSTBRIDGELIB_H
+
+#define BIT(nr)         (1UL << (nr))
+#define GENMASK(h, l) \
+    (((~0UL) - (1UL << (l)) + 1) & (~0UL >> (BITS_PER_LONG - 1 - (h))))
+
```

Comment:
There are definitions, but they are not in the general module.

Reply:
I mean these macros are already defined in a standard header in EDK2. Why not simply use them instead of defining here?

Reply:
I found **BIT** definitions in **edk2** repo:

./NetworkPkg/Library/DxeHttpLib/DxeHttpLib.h:24:#define BIT(x)  (1 << x)
./NetworkPkg/UefiPxeBcDxe/PxeBcDhcp4.h:81:#define BIT(x)   (1 << x)
./ShellPkg/Library/UefiShellDebug1CommandsLib/SmbiosView/QueryTable.c:4009:  #define BIT(Value, bit)  ((Value) & ((UINT32) 1) << (bit))
./ShellPkg/Library/UefiShellDebug1CommandsLib/SmbiosView/PrintInfo.c:23:#define BIT(value, bit)  ((value) & ((UINT64) 1) << (bit))

Found **BIT** definitions in **edk2-platforms** repo
./Silicon/Intel/CoffeelakeSiliconPkg/Cpu/Include/CpuRegs.h:200:#define BIT(Pos, Value)               (1 << (Pos) & (Value))
./Silicon/Intel/KabylakeSiliconPkg/Cpu/Include/CpuRegs.h:239:#define BIT(Pos, Value)               (1 << (Pos) & (Value))
./Silicon/Intel/AlderlakeSiliconPkg/Cpu/Include/CpuRegs.h:41:#define BIT(Pos, Value)               (1 << (Pos) & (Value))
./Silicon/Marvell/Drivers/Net/Pp2Dxe/Mvpp2LibHw.h:12:#define BIT(nr)         (1 << (nr))
./Silicon/Ampere/AmpereAltraPkg/Drivers/PcieInitPei/RootComplexNVParam.h:19:#define BIT(nr) (1 << (nr))
./Silicon/Hisilicon/Drivers/SasV1Dxe/SasV1Dxe.c:135:#define BIT(x)                          (1 << x)
./Platform/Intel/Vlv2TbltDevicePkg/Include/CommonIncludes.h:112:#define BIT(Pos, Value)               (1 << (Pos) & (Value))

No definition of ”BIT“ found in other header files， So I redefined it.

Reply:
Ahh, not sure what I was looking. But it is unfortunate that there are no common definitions of these macros which every one can use :-(.

Still, instead of defining in a platform/vendor specific header, could you please add in generic RISC-V include file under Silicon/RISC-V/Include? That will help in future for any new platforms to avoid duplicating the definition. Agree?

Reply:
Okay，How about putting it in directory “Platform/RISC-V/PlatformPkg/Include/”?/ . And Tested, OK

Reply:
Right. It should be separate patch prior to the commit which uses it.

Reply:
> That will help in future for any new platforms to avoid duplicating the definition. Agree?

I mean, this still makes a basic C macro RiscV-specific. Can we at least simultaneously send a proposal to add this macro to MdePkg Base.h?

Reply:
Thanks Leif. That's what I was initially thinking. But I was not sure about side effects of adding it to Base.h which can create redefinition issues to packages/platforms who might be already having similar definition in different header.  So, I thought we can take that as separate activity (which would require some time to carefully investigate and modify all platforms) instead of making a dependency on this PR.

Reply:
I wasn't really suggesting we make it e dependency, although that is what we *should* do, since we've already between us spent more time discussing it than putting together that PR and submitting it would take.

And I have no sympathy for existing code that decided to add local macros that shouldn't be local.